database/import.go

```diff
@@ -0,0 +1,413 @@
+package database
+
+import (
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"math/big"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/bitcoin-sv/pulse/app/logger"
+	"github.com/bitcoin-sv/pulse/config"
+	"github.com/bitcoin-sv/pulse/database/sql"
+	"github.com/bitcoin-sv/pulse/domains"
+	"github.com/bitcoin-sv/pulse/domains/logging"
+	"github.com/bitcoin-sv/pulse/internal/chaincfg/chainhash"
+	"github.com/bitcoin-sv/pulse/repository"
+)
+
+type DbIndex struct {
+	name string
+	sql  string
+}
+
+type SQLitePragmaValues struct {
+	Synchronous int
+	JournalMode string
+	CacheSize   int
+}
+
+const (
+	insertTransactionSize = 100
+	numWorkers            = 5 // Adjust based on performance and resource constraints
+)
+
+func ImportHeaders(db *sqlx.DB, cfg *config.Config, log logging.Logger) error {
+	log.Info("Import headers from file to the database")
+
+	if !fileExistsAndIsReadable(cfg.Db.PreparedDbFilePath) {
+		return fmt.Errorf("file %s does not exist or is not readable", cfg.Db.PreparedDbFilePath)
+	}
+
+	currentDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	headersRepo := initHeadersRepo(db, cfg)
+
+	dbHeadersCount, _ := headersRepo.GetHeadersCount()
+
+	if dbHeadersCount > 0 {
+		return fmt.Errorf("database already contains %d block headers, the headers table in the database must be empty", dbHeadersCount)
+	}
+
+	compressedHeadersFilePath := filepath.Clean(filepath.Join(currentDir, cfg.Db.PreparedDbFilePath))
+	tmpHeadersFileName := "headers.csv"
+	tmpHeadersFilePath := filepath.Clean(filepath.Join(os.TempDir(), tmpHeadersFileName))
+
+	log.Infof("Decompressing file %s to %s", compressedHeadersFilePath, tmpHeadersFilePath)
+
+	compressedHeadersFile, err := os.Open(compressedHeadersFilePath)
+	if err != nil {
+		return err
+	}
+
+	tmpHeadersFile, err := os.Create(tmpHeadersFilePath)
+	if err != nil {
+		return err
+	}
+
+	if err := gzipDecompressWithBuffer(compressedHeadersFile, tmpHeadersFile); err != nil {
+		return err
+	}
+
+	if err := compressedHeadersFile.Close(); err != nil {
+		return err
+	}
+
+	log.Infof("Decompressed and wrote contents to %s", tmpHeadersFilePath)
+
+	pragmas, err := getSQLitePragmaValues(db)
+	if err != nil {
+		return err
+	}
+
+	if err := modifySQLitePragmas(db); err != nil {
+		return err
+	}
+
+	droppedIndexes, err := removeIndexes(db)
+	if err != nil {
+		return err
+	}
+
+	if err := importHeadersFromFileConcurrent(headersRepo, tmpHeadersFile, log); err != nil {
+		return err
+	}
+
+	if err := tmpHeadersFile.Close(); err != nil {
+		return err
+	}
+
+	if err = restoreIndexes(db, droppedIndexes); err != nil {
+		return err
+	}
+
+	if err = restoreSQLitePragmas(db, *pragmas); err != nil {
+		return err
+	}
+
+	if err := os.Remove(tmpHeadersFilePath); err != nil {
+		return err
+	}
+
+	log.Infof("Deleted temporary file %s", tmpHeadersFilePath)
+
```

Comment:
And here should go some db consistency check.
We could merge this commit without it, but please add the task/subtask for that. Because when we switch to lighter file (without calculated data) this would be even more important to make the check at the end.